src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -301,6 +301,9 @@ public void UpdateTopologyChangeNotification()
 
         private void WaitForLeaderChangeTaskToComplete(Task leaderChangedTask, CancellationTokenSource cts)
         {
+            if (cts == null)
```

Comment:
I think we need to initialize the CTS a bit earlier to avoid getting any possibility of passing `null` here. Also code in the `UpdateTopologyChangeNotification` should use cts.Token all the way down probably. Now it is mixed between cts.Token and ServerShutdown.

Reply:
changed